base/linalg/bidiag.jl

```diff
@@ -263,8 +263,8 @@ end
 transpose(M::Bidiagonal) = Bidiagonal(M.dv, M.ev, !M.isupper)
 ctranspose(M::Bidiagonal) = Bidiagonal(conj(M.dv), conj(M.ev), !M.isupper)
 
-istriu(M::Bidiagonal) = M.isupper || all(M.ev .== 0)
-istril(M::Bidiagonal) = !M.isupper || all(M.ev .== 0)
+istriu(M::Bidiagonal) = M.isupper || all(M.ev .== zero(eltype(M)))
```

Comment:
I'd make this one
```jl
all(t -> t == zero(t), M.ev)
```
because then you use the instance instead of the type to create the zero. Not all types support `zero`, e.g. `Matrix`.